src/components/MultiCombobox/MultiCombobox.tsx

```diff
@@ -157,14 +157,23 @@ function MultiCombobox<Item>({
     },
     [variant]
   );
-  const removeItem = (item: any) => {
+
+  const removeItem = (item: Item) => {
     onChange(value.filter(i => i !== item));
   };
 
-  const addSelectedItem = (item: any) => {
-    if (item !== null) {
-      onChange([...value, item]);
+  const handleChange = (item: Item | Item[] | null) => {
+    if (item === null) {
+      return;
+    }
+    const changedItems = Array.isArray(item) ? item : [item];
+    // If items are already added, remove them from the selected items list
+    if (changedItems.every(item => value.map(itemToString).includes(itemToString(item)))) {
+      return onChange(value.filter(i => !changedItems.map(itemToString).includes(itemToString(i))));
     }
+    // Append non existing items to the list of selected items
+    const newItems = changedItems.filter(i => !value.map(itemToString).includes(itemToString(i)));
+    return onChange([...value, ...newItems]);
```

Comment:
Downshift changes references to the selected items so I cannot use Set here but it would be a great optimization. I had a thought to use `lodash/unionBy` but I opted out.

Reply:
> Downshift changes references to the selected items

Are you 100% sure on that? It sounds bizarre...